mr/src/test/java/org/apache/iceberg/mr/mapred/TestReadMetadataTables.java

```diff
@@ -91,26 +91,34 @@ public void before() throws IOException {
     snapshotId = snapshots.get(0).snapshotId();
   }
 
-  @Ignore("TODO: re-enable this test when snapshot functionality added")
   @Test
   public void testReadSnapshotTable() {
     shell.execute("CREATE DATABASE source_db");
 
     shell.execute(new StringBuilder()
-        .append("CREATE TABLE source_db.table_a ")
-        .append("STORED BY 'org.apache.iceberg.mr.mapred.IcebergStorageHandler' ")
-        .append("LOCATION '")
-        .append(tableLocation.getAbsolutePath() + "/source_db/table_a'")
-        .toString());
+            .append("CREATE TABLE source_db.table_a ")
+            .append("STORED BY 'org.apache.iceberg.mr.mapred.IcebergStorageHandler' ")
+            .append("LOCATION '")
+            .append(tableLocation.getAbsolutePath() + "/source_db/table_a#snapshots'")
+            .toString());
+
+    List<Object[]> result = shell.executeStatement("SELECT * FROM source_db.table_a");
+
+    assertEquals(3, result.size());
```

Comment:
Is there anything we can verify to check that it actually read a snapshot table rather than just a normal table? Are there some fields that are unique to the snapshot who's values we can check?